### PR TITLE
apply clarifications to the gradle documentation based on PiT

### PR DESCRIPTION
**Pull request by taefi**
Fixes: https://github.com/vaadin/flow/issues/10799

### CODE REVIEW

File: articles/guide/start/gradle.asciidoc [hunk 1]
```diff
@@ -96,6 +113,8 @@ The most important files and folders are as follows:
   Though, `gradlew` and `gradle` commands can be used interchangeably if you have Gradle installed already, and you prefer to use your installed Gradle.
   You can find out more about the benefits of using Gradle Wrapper on the https://docs.gradle.org/current/userguide/gradle_wrapper.html[Official Gradle Documentations].
 
+
+
```
Comment (taefi) [inline articles/guide/start/gradle.asciidoc:117]:
```suggestion
```

File: articles/guide/start/gradle.asciidoc [hunk 2]
```diff
@@ -23,16 +23,22 @@ For information about the general usage of Gradle, please refer to the link:http
...
+* Gradle 5 or 6 (Optional, because of the provided wrapper in the starter projects)
+** If you prefer to install it directly, you can find the instructions at https://gradle.org/install
+* Node.js and npm (Optional, because they can also be installed locally to project using the Vaadin Gradle Plugin)
 
 Except for Gradle, these are general Vaadin requirements, as described in <<{articles}/guide/install#,Installing Development Tools>>.
 
+.Installing gradle is optional
+[NOTE]
+If you are going to use Vaadin's Gradle base starter projects, then there is no need to install Gradle on your machine.
+A Gradle Wrapper script is included in those starter projects which will manage to download and run the Gradle locally for your project.
```
Comment (caalador) [inline articles/guide/start/gradle.asciidoc:39]:
```suggestion
A Gradle Wrapper script is included in those starter projects which manage download and execution of Gradle locally for your project.
```

File: articles/guide/start/gradle.asciidoc [hunk 3]
```diff
@@ -331,16 +379,49 @@ vaadin {
...
 ----
 
+.Spring Boot specific configurations
+[NOTE]
+If you are using Vaadin with Spring Boot, normally the default packaging for production would be the `jar`.
+But, if you intend to package a Spring Boot application as WAR to be deployed on a standalone container such as `tomcat`, there are two additional steps needed to take:
+
+* Your application class that is annotated with `@SpringBootApplication` should extend `SpringBootServletInitializer` and override the `configure` method:
+
+.Example of enabling Spring Boot servlet initializer
```
Comment (caalador) [inline articles/guide/start/gradle.asciidoc:394]:
```suggestion
.Example of enabling SpringBootServletInitializer
```

File: articles/guide/start/gradle.asciidoc [hunk 4]
```diff
@@ -331,16 +379,49 @@ vaadin {
...
+== Using Gradle Plugin Snapshot Versions
 
 A snapshot version of the plugin is pushed to the pre-release repository.
+Note that this section is about trying the pre-release and snapshot versions of the Vaadin Gradle Plugin itself, not the Vaadin platform.
 
 To use the pre-released plugin add the `vaadin-prereleases` repository to the project `settings.gradle` file.
+`settings.gradle` is mostly used within multi-module projects, but it comes handy for other configurations as well.
+Thus, if you don't have it already in your project, go ahead and create a plain text file called `settings.gradle` next to your `build.gradle` file (normally in project root folder).
 
 .Plugin repository added to settings file
```
Comment (caalador) [inline articles/guide/start/gradle.asciidoc:426]: Perhaps add a reminder to remove the `id 'com.vaadin' version 'xyz'` from the plugins block.

**Review by caalador — CHANGES_REQUESTED** [review]

**Review by caalador — APPROVED** [review]

**Pull request merged**